src/storage/src/hummock/compactor/compactor_runner.rs

```diff
@@ -239,13 +243,26 @@ impl CompactorRunner {
 
         // The `SkipWatermarkIterator` is used to handle the table watermark state cleaning introduced
         // in https://github.com/risingwavelabs/risingwave/issues/13148
-        Ok(SkipWatermarkIterator::from_safe_epoch_watermarks(
-            MonitoredCompactorIterator::new(
-                MergeIterator::for_compactor(table_iters),
-                task_progress.clone(),
-            ),
-            &self.compact_task.table_watermarks,
-        ))
+        let combine_iter = {
```

Comment:
Let's add comment and referencing issue for state cleaning based on NonPkPrefix watermark iterator as well.